messages/teleporter/message_handler.go

```diff
@@ -167,7 +183,47 @@ func (m *messageHandler) ShouldSendMessage(destinationClient vms.DestinationClie
 		return false, nil
 	}
 
-	return true, nil
+	var decision bool = true
```

Comment:
Should this be initialized to false? It looks like the error cases are just returning decision without modifying it, and I assume we should return false in those cases.

Reply:
The intended control flow here is to separate the existing `ShouldSendMessage` decision logic from the `Decider` logic that is dispatched to the service. If the `Decider` fails, then we should return the prior decision from the existing logic.

I think this could be made clearer by adding a new helper, and calling it like so:

```
ShouldSendMessage() {
  // Existing decision logic
  decision, err :=  m.dispatchToDecider()
  if err != nil
    // log the error, but do not return it
    log.Warn("Decider returned error")
    return true, nil  
  }
  return decision
}
```

Reply:
I extracted the logic into a new helper, and adapted my changes to `ShouldSendMessage` to fit the existing flow logic (if __ return false; if ___ return false; ...; return true), in https://github.com/ava-labs/awm-relayer/pull/344/commits/584556eccf9ea1c2fa23db5adffdb15b5d210fd4